File: bindings/lua/Polycode/AttributeBinding.lua
class "AttributeBinding"


function AttributeBinding:__getvar(name)
	if name == "name" then
		return Polycode.AttributeBinding_get_name(self.__ptr)
	elseif name == "enabled" then
		return Polycode.AttributeBinding_get_enabled(self.__ptr)
	end
end

function AttributeBinding:__setvar(name,value)
	if name == "name" then
		Polycode.AttributeBinding_set_name(self.__ptr, value)
		return true
	elseif name == "enabled" then
		Polycode.AttributeBinding_set_enabled(self.__ptr, value)
		return true
	end
	return false
end
function AttributeBinding:AttributeBinding(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.AttributeBinding(unpack(arg))
	end
end

function AttributeBinding:__delete()
	if self then Polycode.delete_AttributeBinding(self.__ptr) end
end

File: bindings/javascript/Polycode/AttributeBinding.js
function AttributeBinding() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.AttributeBinding()
	}
	Object.defineProperties(this, {
		'name': { enumerable: true, configurable: true, get: AttributeBinding.prototype.__get_name, set: AttributeBinding.prototype.__set_name},
		'enabled': { enumerable: true, configurable: true, get: AttributeBinding.prototype.__get_enabled, set: AttributeBinding.prototype.__set_enabled}
	})
}


AttributeBinding.prototype.__get_name = function() {
	return Polycode.AttributeBinding__get_name(this.__ptr)
}

AttributeBinding.prototype.__set_name = function(val) {
	Polycode.AttributeBinding__set_name(this.__ptr, val)
}

AttributeBinding.prototype.__get_enabled = function() {
	return Polycode.AttributeBinding__get_enabled(this.__ptr)
}

AttributeBinding.prototype.__set_enabled = function(val) {
	Polycode.AttributeBinding__set_enabled(this.__ptr, val)
}

Duktape.fin(AttributeBinding.prototype, function (x) {
	if (x === AttributeBinding.prototype) {
		return;
	}
	Polycode.AttributeBinding__delete(x.__ptr)
})
